zerver/tests/test_channel_permissions.py

```diff
@@ -0,0 +1,729 @@
+import orjson
+from typing_extensions import override
+
+from zerver.actions.realm_settings import (
+    do_change_realm_permission_group_setting,
+    do_set_realm_property,
+)
+from zerver.actions.streams import do_change_stream_group_based_setting
+from zerver.actions.user_groups import add_subgroups_to_user_group, check_add_user_group
+from zerver.actions.users import do_change_user_role
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.test_helpers import get_subscription
+from zerver.lib.types import UserGroupMembersData
+from zerver.models import NamedUserGroup, Recipient, Stream, Subscription, UserProfile
+from zerver.models.groups import SystemGroups
+from zerver.models.realms import get_realm
+from zerver.models.streams import get_stream
+
+
+class ChannelSubscriptionPermissionTest(ZulipTestCase):
```

Comment:
Have used "Channel" here instead of "Stream" since this is a new class, and in other commits whole class have been moved, so have not changed that. Not sure if this can be confusing because we still use "stream" at other places.